internal/component/pyroscope/receive_http/receive_http.go

```diff
@@ -0,0 +1,182 @@
+package receive_http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/grafana/alloy/internal/component/pyroscope/write"
+	"golang.org/x/sync/errgroup"
+	"io"
+	"net/http"
+	"reflect"
+	"sync"
+
+	"github.com/grafana/alloy/internal/component"
+	fnet "github.com/grafana/alloy/internal/component/common/net"
+	"github.com/grafana/alloy/internal/component/pyroscope"
+	"github.com/grafana/alloy/internal/featuregate"
+	"github.com/grafana/alloy/internal/runtime/logging/level"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "pyroscope.receive_http",
+		Stability: featuregate.StabilityGenerallyAvailable,
```

Comment:
Should we start with Public Preview? GA means we have to follow backwards compatibility and performance needs.